apache/superset
# feat: customize recent activity access

### SUMMARY
Adds a config flag `ENABLE_BROAD_ACTIVITY_ACCESS` that makes it possible to disable access to other users' recent activity data, both via the API and the profile page. In addition, a security manager method `raise_for_user_activity_access` is introduced to make it possible to fine tune who can and can't access recent activity data.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
When the flag is set to false, trying to access other users' activity data returns a 403 which is displayed on the profile page tables with the message "Access to user activity data is restricted" (in the screenshot the user is viewing another user's profile page):
![image](https://user-images.githubusercontent.com/33317356/144017932-d76b7604-d514-4f18-998a-9fd0d9a8d59f.png)

With the default settings, accessing other users' activity data is allowed (current behavior unchanged):
![image](https://user-images.githubusercontent.com/33317356/144018164-714df161-74b9-4444-aba8-17d5acc62064.png)

### TESTING INSTRUCTIONS
1. set `ENABLE_BROAD_ACTIVITY_ACCESS = False` in `superset_config.py`
2. login as `admin`
3. create a user `test`
3. go to `/superset/profile/test`
4. notice the message "Access to user activity data is restricted" in all activity tables
5. comment out `ENABLE_BROAD_ACTIVITY_ACCESS` flag in `superset_config.py`
6. go to `/superset/profile/test`
7.  notice that access is unrestricted

### ADDITIONAL INFORMATION
